### PR TITLE
[RSDK-2734] turn off PWM signals when closing a board

### PR DESCRIPTION
**Pull request by penguinland**
Tried on an Orin: if you set a pin to high _or_ turn on a PWM signal, and then hit control-C to kill the RDK server, the pin goes low again and stays there. 

If you set a pin high or turn on a PWM signal, and then change the board config, the pin goes low again and stays there (until you set it to something else afterwards on the new board component). Is that the right choice? I think it might be, but would like a second opinion.

### CODE REVIEW

File: components/board/genericlinux/gpio.go [hunk 1]
```diff
@@ -280,6 +280,12 @@ func (pin *gpioPin) Close() error {
 	pin.mu.Lock()
 	defer pin.mu.Unlock()
 
+	// If the entire server is shutting down, it's important to turn off all pins so they don't
+	// continue outputting signals we will no longer control.
+	if err := pin.setInternal(false); err != nil {
+		return err
+	}
```
Comment (randhid) [inline components/board/genericlinux/gpio.go:287]: [q] this should always be called before line 289? Or does it not matter when we unexport our definitions before setting states up or down?

**Review by randhid — APPROVED** [review]
LGTM - one question, if that affects anything please change it.

File: components/board/genericlinux/gpio.go (hunk 1 above)
Comment (penguinland) [reply components/board/genericlinux/gpio.go:287]: It doesn't matter, but double-checking that it doesn't matter made me notice that `setInternal` calls `openGpioFd`, which contains a copy of lines 289-294 (when opening the GPIO interface, we close the PWM interface). I'll remove the duplicate code below.

**Comment by penguinland** [comment]
Yup, even with the latest version (after removing the duplicate code), we've got the same behavior described in the PR description.

**Pull request merged**